packages/notion-dump/vite.config.ts

```diff
@@ -0,0 +1,23 @@
+import { defineConfig } from "vite";
+import { resolve } from "path";
+
+export default defineConfig({
+  build: {
+    lib: {
+      entry: resolve(__dirname, "src/lib/index.ts"),
+      name: "NotionDump",
+      fileName: (format) => `notion-dump.${format}.js`,
+    },
+    rollupOptions: {
+      external: ["@cozy-blog/notion-client", "commander", "path", "fs"],
+      output: {
+        globals: {
+          "@cozy-blog/notion-client": "NotionClient",
+          commander: "commander",
+          path: "path",
+          fs: "fs",
+        },
+      },
+    },
+  },
+});
```

Comment:
p4,  Since configurations like Vite config are often the most challenging part in open-source projects, it would be helpful to add comments explaining the intention and purpose behind them.

ask; It seems like this project is using TypeScript, but there doesn't appear to be any TypeScript configuration in the Vite setup. (The TypeScript settings below were generated by Claude, so they might not be entirely accurate.) Is there any reason for the missing TypeScript configuration? If there is relevant context, please explain it through comments.


```suggestion
mport { defineConfig } from "vite";
import { resolve } from "path";

// Library name
const LIB_NAME = "NotionDump";

export default defineConfig({
  build: {
    // Library specific build configuration
    lib: {
      // Set the entry point of the library
      entry: resolve(__dirname, "src/lib/index.ts"),
      name: LIB_NAME,
      fileName: (format) => `${LIB_NAME.toLowerCase()}.${format}.js`,
    },
    rollupOptions: {
      // Specify external dependencies not to be bundled
      external: ["@cozy-blog/notion-client", "commander", "path", "fs"],
      output: {
        // Global variable mapping for UMD build
        globals: {
          "@cozy-blog/notion-client": "NotionClient",
          commander: "commander",
          path: "path",
          fs: "fs",
        },
      },
    },
  },

  // TypeScript settings
  plugins: [
    {
      name: 'typescript-plugin',
      buildStart() {
        // Load tsconfig.json file (assuming it's in the project root)
        const tsConfig = require('./tsconfig.json');
        // Merge tsconfig.json settings into Vite's TypeScript configuration
        this.config.esbuild = {
          ...this.config.esbuild,
          ...tsConfig.compilerOptions,
        };
      },
    },
  ],
});
```